common/schema.go

```diff
@@ -0,0 +1,89 @@
+package common
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"github.com/xeipuuv/gojsonschema"
+)
+
+const schema = `{
+    "title": "V2 func file schema",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type":"string"
+        },
+        "schema_version": {
+            "type":"integer"
+        },
+        "version": {
+            "type":"string"
+        },
+        "runtime": {
+            "type":"string"
+        },
+        "build_image": {
+            "type":"string"
+        },
+        "run_image": {
+            "type": "string"
+        },
+        "entrypoint": {
+            "type":"string"
+        },
+        "content_type": {
+            "type":"string"
+        },
+        "cmd": {
+            "type":"string"
+        },
+        "memory": {
+            "type":"integer"
+        },
+        "timeout": {
+            "type":"integer"
+        },
+        "idle_timeout": {
+            "type": "integer"
+        },
+        "config": {
+            "type": "integer"
+        },
+        "triggers": {
+            "type": "array",
+            "properties": {
+                "name": {
+                    "type":"string"
+                },
+                "type": {
+                    "type":"string"
+                },
+                "source": {
+                    "type":"string"
+                }
+            }
+        }
+    }
+}`
+
+func ValidateSchema(jsonFile string) error {
```

Comment:
does this validate the schema? or does it actually validate the file?

Consider breaking this out so the schema is an arg.

Reply:
It validates the file against the schema. Or are you referring to the name of the function?